docs/tutorial/typesfuns.rst

```diff
@@ -1182,3 +1183,30 @@ matching ``let`` and lambda bindings. It will *only* work if:
 - The type of the result is "known". i.e. the type of the expression
   can be determined *without* type checking the ``case``-expression
   itself.
+
+Totality
+========
+
+A *total* function is one that is defined for all inputs of the correct
+type: that is, it is guaranteed not to crash. For example, a function
+that squares an integer is total for all integers:
+
+.. code-block:: idris
+
+    square : Int -> Int
+    square x = x * x
+
+However, the ``head`` function is *not* total; it will error out when
+called on an empty list:
+
+.. code-block:: idris
+
+    Idris> head [1, 2, 3]
```

Comment:
Also, it would be nice to show `magic`  above as an example of a covering but diverging function, I believe.